crates/abq_queue/src/job_queue.rs

```diff
@@ -212,12 +235,16 @@ mod test {
                 chunks.push((entity, 1));
             }
         }
+
+        // Now, go through the queue's assigned entities and make sure everything looks okay.
+        // There should be no holes, and the runs of assignments should align with how many tests
+        // each worker popped off.
         let mut chunks_it = chunks.into_iter().peekable();
         while let Some((entity, run)) = chunks_it.next() {
             let entity_batch = workers.get(entity.0.get()).unwrap();
             match chunks_it.peek() {
-                Some(_) => assert_eq!(run, entity_batch.get()),
-                None => assert!(run <= entity_batch.get()),
+                Some(_) => assert_eq!(run, entity_batch.get()), // if there are more chunks, batch size should be full
```

Comment:
This may be better in the assertion
```suggestion
                Some(_) => assert_eq!(run, entity_batch.get(), "if there are additional chunks, the entity should have pulled exactly batch_size"),
```